mongomock/collection.py

```diff
@@ -416,6 +416,11 @@ def _update(self, spec, document, upsert=False, manipulate=False,
                     subdocument = self._update_document_fields_with_positional_awareness(
                         existing_document, v, spec, updater, subdocument)
 
+                elif k == '$rename':
+                    for src, dst in iteritems(v):
+                        if self._has_key(existing_document, src):
```

Comment:
I think you don't handle the case where `src` or `dst` contains a dot. You can either try to implement it or check if `src` or `dst` contains a dot and raise a `NotImplementedError` if it's the case, explaining that this is a valid operation but that `mongomock` does not handle it yet.